cmd/service_linux.go

```diff
@@ -0,0 +1,316 @@
+		err = systemctlEnableCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+		return nil
+	},
+}
+
+var uninstallCmd = &cobra.Command{
+	Use:               "uninstall",
+	Short:             "Uninstall the startup process for Cloudfuse. Requires running as admin.",
+	Long:              "Uninstall the startup process for Cloudfuse. Requires running as admin.",
+	SuggestFor:        []string{"uninst", "uninstal"},
+	Example:           "cloudfuse service uninstall",
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		// 1. find and remove cloudfuse.service from /etc/systemd/system and run systemctl daemon-reload
+		removeFileCmd := exec.Command("sudo", "rm", "/etc/systemd/system/cloudfuse.service")
+		err := removeFileCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to delete cloudfuse.service file from /etc/systemd/system due to following error: [%s]", err.Error())
+		}
+
+		systemctlDaemonReloadCmd := exec.Command("sudo", "systemctl", "daemon-reload")
+		err = systemctlDaemonReloadCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+
+		return nil
+	},
+}
+
+//--------------- command section ends
+
+func collectServiceData(serviceFilePath string) (map[string]string, error) {
+	serviceFile, err := os.Open("./setup/cloudfuse.service")
+
+	if err != nil {
+		fmt.Println("Error opening file:", err)
+		return nil, err
+	}
+
+	defer serviceFile.Close()
+
+	scanner := bufio.NewScanner(serviceFile)
+	serviceData := make(map[string]string)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, "User=") {
+			parts := strings.SplitN(line, "=", 2)
+			key := strings.TrimSpace(parts[0])
+			value := strings.TrimSpace(parts[1])
+			serviceData[key] = value
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		fmt.Println("Error reading file:", err)
+		return nil, err
+	}
+	return serviceData, nil
+}
+
+func newSericeFile(mountPath string, configPath string, dir string) (string, error) {
+
+	//mountpath or config?
+	var oldString string
+	var newString string
+
+	oldConfigStr := "Environment=ConfigFile=/path/to/config/file/config.yaml"
+	newConfigStr := fmt.Sprintf("Environment=ConfigFile=%s", common.JoinUnixFilepath(dir, configPath))
+
+	oldMountStr := "Environment=MoutingPoint=/path/to/mounting/point"
+	newMountStr := fmt.Sprintf("Environment=MoutingPoint=%s", common.JoinUnixFilepath(dir, mountPath))
+
+	// open service defaultFile for read write
+	defaultFile, err := os.OpenFile("./setup/cloudfuse.service", os.O_RDWR, 0644)
+	if err != nil {
+		return "", fmt.Errorf("error opening file: [%s]", err.Error())
+	}
+	defer defaultFile.Close()
+
+	scanner := bufio.NewScanner(defaultFile)
+	var lines []string
+
+	// Read the file line by line
+	for scanner.Scan() {
```

Comment:
Rather than doing string replacement, why not just have a template for the service file you would like to write, and then add the configFile and mountPath information to that template and then write the template file? Possibly using text/template or something similar.

Reply:
I am using the setup/cloudfuse.service file as a template. you mentioned in another comment about how this file may not exist if the installer was used. How would you suggest I have a template reference in that scenario? Should I have the template hard coded and reference it as needed?

Reply:
Yeah I would not rely on that file existing. The executable should be able to write the file to systemd. So like suggested above I would have that hard coded in a string or template and then write the file.